### PR TITLE
allow loading/saving netdata dashboards

### PR DESCRIPTION
**Pull request by ktsaou**
This PR adds support these cases:

- [x] exporting a netdata dashboard as PDF (print to PDF). This is useful for sending the dashboards to a third party, not familiar with netdata and without access to a netdata.

- [x] exporting a netdata dashboard as JSON. This is useful for maintaining snapshots of the entire dashboard (all charts and their values for the visible timeframe).

- [x] importing a netdata dashboard from JSON. So, dashboards exported as JSON can be loaded back (to any netdata - no need to find the same one). Once loaded the dashboard is fully interactive, including pan and zoom operations. Of course the detail depends on the exported data.

---

### Why?

Well... we use netdata dashboards as evidence of issues. In many cases dashboards have to be shared with vendors, hosting providers, management, etc.

Printing to PDF allows anyone to view netdata dashboards. They can even be attached to legal documents.

Exporting and importing dashboards in JSON, allows talking snapshots that can loaded back to examine new findings, or validate new assumptions.

---

### Exporting to PDF

image

Once  :printer: (print)  is clicked a modal comes up:

image

Clicking **Print** opens a new window, in white, that downloads and renders all the charts:

peek 2017-11-11 19-55

When all the charts are ready, it pops-up the print dialog:

screenshot from 2017-11-11 19-32-20

And here is the final result (10MB):

[ktsaou-laptop netdata dashboard.pdf](https://github.com/firehol/netdata/files/1463937/ktsaou-laptop.netdata.dashboard.pdf)

Keep in mind in portrait printing the detail is not as good as in landscape mode (more width = more detail at the charts).

Special care has been given to:

1. resize all the charts so that all the legends are visible
2. the current pan and zoom settings of the dashboard are respected (so, just align the charts on the dashboard before hitting **Print** to have the exact same view printed)
3. instruct the browser to avoid splitting a chart into multiple pages
4. remove all charts that show just a snapshot of the data (gauge charts and easypiecharts)

PDFs are better that images, because they are searchable... Unfortunately, not as detailed...

### Exporting Snapshots

Snapshot settings:

image

Exporting progress:

image

### Importing Snapshots

screenshot from 2017-11-13 12-45-42

And here is the whole process (export + import):

peek 2017-11-13 12-58

### CODE REVIEW

**Comment by ktsaou** [comment]
All new features implemented.
Testers wanted.

(loading and saving netdata snapshots is amazing!)

The only problem is that firefox does not save them. Probably my fault. On all other browsers it works.

**Comment by ktsaou** [comment]
fixed firefox too.

**Comment by l2isbad** [comment]
Icons arrow direction looks wrong for me. Export looks like import, import like export.

File: web/index.html [hunk 1]
```diff
@@ -3120,12 +3692,15 @@
...
-                    <li><a href="https://www.facebook.com/linuxnetdata/" class="btn" target="_blank" title="netdata on facebook"><i class="fa fa-facebook-official" aria-hidden="true"></i></a></li>
+                    <li class="hidden-sm hidden-md"><a href="https://github.com/firehol/netdata/wiki" class="btn" target="_blank" title="netdata on github"><i class="fa fa-github"></i></a></li>
+                    <li class="hidden-sm hidden-md"><a href="https://twitter.com/linuxnetdata" class="btn" target="_blank" title="netdata on twitter"><i class="fa fa-twitter" aria-hidden="true"></i></a></li>
+                    <li class="hidden-sm hidden-md"><a href="https://www.facebook.com/linuxnetdata/" class="btn" target="_blank" title="netdata on facebook"><i class="fa fa-facebook-official" aria-hidden="true"></i></a></li>
+                    <li id="loadButton"><a href="#" class="btn" data-toggle="modal" data-target="#loadSnapshotModal" title="load a saved dashboard snapshot"><i class="fa fa-upload"></i>&nbsp;<span class="hidden-sm hidden-md hidden-lg">Load</span></a></li>
```
Comment (ktsaou) [inline web/index.html]: @l2isbad I used `upload` for `load`

File: web/index.html [hunk 2]
```diff
@@ -3120,12 +3692,15 @@
...
+                    <li id="saveButton"><a href="#" class="btn" data-toggle="modal" data-target="#saveSnapshotModal" title="save a snapshot of this dashboard"><i class="fa fa-download"></i>&nbsp;<span class="hidden-sm hidden-md hidden-lg">Save</span></a></li>
```
Comment (ktsaou) [inline web/index.html]: and `download` for `save`

**Comment by ktsaou** [comment]
I guess the arrows have as reference the users computer. So, into the computer means download and out of the computer mean upload.

**Comment by l2isbad** [comment]
> I guess the arrows have as reference the users computer.

Yes, i know it. But reference to `netdata` is more intuitive for me. Since it's netdatas dahsboard and actions are export dashboard/import dashboard.

**Comment by ktsaou** [comment]
hm... the other way would by to use the open folder and floppy disk for load and save. Check this http://fontawesome.io/icons/ - I am open to ideas...

Since this PR is purely javascript, I think it is safe to merge it, so that people will start using it and report issues. What do you think?

**Comment by l2isbad** [comment]
> floppy disk

😄

**Comment by ktsaou** [comment]
hm... this is not so nice:

image

**Comment by l2isbad** [comment]
My suggestion was to use `download` for import, `upload` for export.

**Comment by ktsaou** [comment]
here you are...

**Comment by ktsaou** [comment]
I renamed Load to Import and Save to Export to match the flow if the icons.

**Comment by ktsaou** [comment]
So, I will merge this as soon as travis completes...
Happy snapshoting !

**Comment by ktsaou** [comment]
peek 2017-11-13 13-13

**Comment by ktsaou** [comment]
merged it !

**Pull request merged**